.travis.yml

```diff
@@ -1,20 +1,29 @@
 sudo: false
 language: python
+before_install: sudo apt-get install python-tk python3-tk ghostscript
+# command to install dependencies
+install: "make"
 cache: pip
-python:
-  - "2.7"
-  - "3.5"
-  - "3.6"
-matrix:
+jobs:
   include:
-    - python: 3.7
+    - stage: test
+      script:
+        - make test
+      python: '2.7'
+    - stage: test
+      script:
+        - make test
+      python: '3.5'
+    - stage: test
+      script:
+        - make test
+      python: '3.6'
+    - stage: test
+      script:
+        - make test
+      python: '3.7'
       dist: xenial
-      sudo: true
-before_install:
-  - sudo apt-get install python-tk python3-tk ghostscript
-install:
-  - pip install ".[dev]"
-script:
-  - pytest --verbose --cov-config .coveragerc --cov-report term --cov-report xml --cov=camelot tests
-after_success:
-  - codecov --verbose
\ No newline at end of file
+      sudo: true     
+    - stage: coverage
+      python: 3.6
+      script: make test && codecov --verbose
```

Comment:
@vaibhavmule The requests Makefile just calls codecov at this stage? I see in the docs that files aren't preserved between stages, any idea how they get the coverage.xml to upload at this stage?

Reply:
They (Requests) don't, their coverage thing is broken.

Reply:
Oh didn't know that. Any idea why?

Reply:
I'm not sure, but as far as I know, codecov requires the tests to be run and coverage file to be generated when you want send things to codecov. 

This is how we do at @MasoniteFramework https://github.com/MasoniteFramework/core/blob/develop/.travis.yml#L13

Reply:
Yeah just not a big fan of the `&&` :) Let's keep it for now till we can think of a nicer way. Maybe adding a directive in the Makefile will reduce this to a single command.

Reply:
if not `&&`, does this make sense for you?

```
script: 
  - make test
  - codecov --verbose
```